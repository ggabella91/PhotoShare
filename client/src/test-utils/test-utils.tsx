import { FC, ReactElement } from 'react';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import { render, RenderOptions } from '@testing-library/react';
import { store } from '../redux/store';

interface WrapperProps {
  wrapperProps?: { route: string; location: string };
}

type CustomRenderOptions = RenderOptions & WrapperProps;

const WithReduxProvider: FC<WrapperProps> = ({ children, wrapperProps }) => {
  return (
    <Provider store={store}>
      <BrowserRouter>
        <Routes>
          <Route path={wrapperProps?.route || '/'} element={children} />
        </Routes>
      </BrowserRouter>
    </Provider>
  );
};

const renderWithReduxAndRouter = (
  ui: ReactElement,
  options?: Omit<CustomRenderOptions, 'wrapper'>
) => {
  window.history.pushState({}, '', options?.wrapperProps?.location || '/');
  const customContainer = document.createElement('div');
  customContainer.id = 'root';

  return render(ui, {
    wrapper: (props) => (
      <WithReduxProvider {...props} wrapperProps={options?.wrapperProps} />
    ),
    container: document.body.appendChild(customContainer),
    ...options,
  });
};

export * from '@testing-library/react';
export { default as userEvent } from '@testing-library/user-event';
export { renderWithReduxAndRouter as render };
