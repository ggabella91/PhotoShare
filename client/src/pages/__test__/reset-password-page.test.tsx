import { render, screen, userEvent } from '../../test-utils/test-utils';
import { ResetPasswordPage } from '../reset-password/reset-password-page.component';

describe('reset password page component tests', () => {
  const setup = () => {
    console.error = jest.fn();

    render(<ResetPasswordPage />, {
      wrapperProps: {
        route: '/reset-password/:token',
        location: '/reset-password/dasijgfsdhnag',
      },
    });
  };

  it('renders a reset password page component', () => {
    setup();

    const resetPasswordPage = screen.getByText('Set your new password below.');

    expect(resetPasswordPage).toBeInTheDocument();
  });

  it("clicking 'change password' causes resetPasswordStart action creator to be called without throwing an error", () => {
    setup();

    const password = screen.getByLabelText('password');
    const confirmPassword = screen.getByLabelText('confirm password');
    const submitButton = screen.getByText('Change Password');

    userEvent.type(password, 'dis my new password');
    userEvent.type(confirmPassword, 'dis my new password');
    userEvent.click(submitButton);

    expect(console.error).not.toHaveBeenCalled();
  });
});
