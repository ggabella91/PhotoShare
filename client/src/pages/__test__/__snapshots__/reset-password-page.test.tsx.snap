// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a reset password page component 1`] = `
<Router
  history={
    Object {
      "action": "POP",
      "block": [Function],
      "canGo": [Function],
      "createHref": [Function],
      "entries": Array [
        Object {
          "hash": "",
          "key": "f4blcg",
          "pathname": "/reset-password/sdvb7bgiy4g3487ew",
          "search": "",
          "state": undefined,
        },
      ],
      "go": [Function],
      "goBack": [Function],
      "goForward": [Function],
      "index": 0,
      "length": 1,
      "listen": [Function],
      "location": Object {
        "hash": "",
        "key": "f4blcg",
        "pathname": "/reset-password/sdvb7bgiy4g3487ew",
        "search": "",
        "state": undefined,
      },
      "push": [Function],
      "replace": [Function],
    }
  }
>
  <Route
    path="/reset-password/:token"
  >
    <ResetPasswordPage
      resetConfirm={null}
      resetError={null}
      resetPasswordStart={[Function]}
    />
  </Route>
</Router>
`;
